processing/src/main/java/org/apache/druid/segment/IndexIO.java

```diff
@@ -643,37 +649,38 @@ public QueryableIndex load(File inDir, ObjectMapper mapper, boolean lazy, Segmen
           smooshedFiles,
           loadFailed
       );
+      // todo (clint): just pass in the metadata to SimpleQueryableIndex so it doesnt need to read it again...
+      final Map<String, Map<String, Supplier<ColumnHolder>>> projectionsColumns = new LinkedHashMap<>();
+      final Metadata metadata = getMetdata(smooshedFiles, mapper, inDir);
+      if (metadata != null && metadata.getProjections() != null) {
```

Comment:
Is there somewhere convenient to register these to save on heap space? I would think that lots of segments on the same server are going to have exactly the same projection spec. And the projection specs could add up in size if we store each copy separately. Figure 100–200 bytes for each grouping column and each aggregation, multiplied by possibly 10–100 columns, multiplied by 100k+ segments on a big server. It could believably be multi-GB if the projections have a lot of columns.

Reply:
yea, will look into interning it. Ordering probably needs this too so will also look into that

Reply:
done, ordering is also interned in SimpleQueryableIndex